optimize size and time using "--no-cache-dir"

rajpratik71 opened the pull request:
Using "--no-cache-dir" flag in pip install ,make sure dowloaded packages
by pip don't cached on system . This is a best practise which make sure
to fetch ftom repo instead of using local cached one . Further , in case
of Docker Containers , by restricing caching , we can reduce image size.
In term of stats , it depends upon the number of python packages
multiplied by their respective size . e.g for heavy packages with a lot
of dependencies it reduce a lot by don't caching pip packages.

Further , more detail information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

kevin-bates commented on all-spark-kernels/Dockerfile:
```diff
@@ -6,7 +6,7 @@ FROM jupyter/all-spark-notebook:97a5071c5775
 
 # Install Kernel Gateway
-RUN pip install 'jupyter_kernel_gateway>=1.0,<2.0'
+RUN pip install --no-cache-dir 'jupyter_kernel_gateway>=1.0,<2.0'
```
Let's drop the restrictions on JKG version.

rajpratik71 replied:
Done as requested

kevin-bates commented on node_client_example/Dockerfile.kg:
```diff
@@ -5,7 +5,7 @@
 
 # install the kernel gateway
-RUN pip install jupyter_kernel_gateway==0.5.*
+RUN pip install --no-cache-dir jupyter_kernel_gateway==0.5.*
```
JKG version.

rajpratik71 replied:
Done as requested

kevin-bates commented on python_client_example/Dockerfile.kg:
```diff
@@ -5,7 +5,7 @@
 
 # install the kernel gateway
-RUN pip install 'jupyter_kernel_gateway==0.5'
+RUN pip install --no-cache-dir 'jupyter_kernel_gateway==0.5'
```
Ditto - drop version restriction.

rajpratik71 replied:
Done as requested

kevin-bates commented on swagger-notebook-service/jupyter-swagger-codegen/src/main/resources/jupyter-swagger-codegen/Dockerfile.mustache:
```diff
@@ -3,7 +3,7 @@
 
 # Install Kernel Gateway
-RUN pip install git+https://github.com/jupyter-incubator/kernel_gateway.git
+RUN pip install --no-cache-dir git+https://github.com/jupyter-incubator/kernel_gateway.git
```
Convert to straight pip install like the others - incubator is no longer required.

rajpratik71 replied:
Done as requested

kevin-bates commented on swagger-notebook-service/swagger-petstore-service/Dockerfile:
```diff
@@ -3,7 +3,7 @@
 
 # Install Kernel Gateway
-RUN pip install git+https://github.com/jupyter-incubator/kernel_gateway.git
+RUN pip install --no-cache-dir git+https://github.com/jupyter-incubator/kernel_gateway.git
```
ditto previous.

rajpratik71 replied:
Done as requested

kevin-bates reviewed (commented):
@Rajpratik71 - thank you for the contribution - this looks good.  As you can see from my comments, you've stumbled upon some very out of date code and we should go ahead and take the opportunity to make the appropriate updates.  Thank you.

rajpratik71 committed:
optimize size and time using "--no-cache-dir"

Some version restriction dropped and  further Using "--no-cache-dir"
flag in pip install ,make sure dowloaded packages
by pip don't cached on system . This is a best practise which make sure
to fetch ftom repo instead of using local cached one . Further , in case
of Docker Containers , by restricing caching , we can reduce image size.
In term of stats , it depends upon the number of python packages
multiplied by their respective size . e.g for heavy packages with a lot
of dependencies it reduce a lot by don't caching pip packages.

Further , more detail information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

kevin-bates reviewed (approved):
LGTM - thank you!
